play.py

```diff
@@ -89,93 +90,91 @@ def computerPlay(dict_name, chosenCategory, secretLetter, secretWord):
+            print("I am out of words")
+            break
+        else:
+            wordGuess = random.choice(new_words_available)
+            usedWords.append(wordGuess)
 
-                    break
-                else:
-                    pass
+        print('Is your secret word ' + str(wordGuess) + ' ?')
+        time.sleep(2)
+        userResponse = input()
 
-            else:
-                print("Apologies, I am unable to guess your word")
-                
-                # user should provide word that begins with same starting letter specified earlier
-                while True:
-                    print("What was your word?")
-                    userWord = input().lower()
+        if userResponse == "yes" or userResponse == 'y':
+            print("Hooray, I made the right guess in " + str(computerGuesses) + " attempt(s)")
+            return True
 
-                    if userWord[0] == userStartingLetter:
-                        userWords.append(userWord)
+    print("Apologies, I am unable to guess your word")
+    return False
 
-                        break
-                else:
-                    print(f"Your secret word must begin with the letter {userStartingLetter} you specified earlier") 
-            
+
+# Function for the user to provide their word
+def user_provide_word(userStartingLetter, userWords):
+    while True:
+        print("What was your word?")
+        userWord = input().lower()
+
+        if userWord[0] == userStartingLetter:
+            userWords.append(userWord)
             break
         else:
-            unavailableKey +=1
-
-            print()
-            print(f"My database of {dict_name} does not have words that begin {userStartingLetter}")
-            print("Please choose another letter")
+            print(f"Your secret word must begin with the letter {userStartingLetter} you specified earlier")
 
-    else:
-        print("You have entered an unavailable letter too many times")
+# main function to control the user's turn
+def userPlay(dict_name, chosenCategory, userWords=[]):
 
+    # userWords list is for words the computer fails to guess and are provided by the user
+    # if the user continues to play, this list of words should be available and appended to
 
+    print()
 
+    if dict_name != "expert":
+        print(f"It is your turn to play. Please type in the first letter of your secret {dict_name[:-1]}")
+    else:
+        print("It is your turn to play. Please type in the first letter of your secret object")
 
+    print('Type "yes / y" or "no / n" to my guesses')
+    print()
 
+    userStartingLetter = get_user_starting_letter(chosenCategory)
+    
+    usedWords = []
+    
+    if userStartingLetter:
+        while True:
+            result = computer_guess_words(chosenCategory, userStartingLetter, usedWords)
 
+            if result:
+                break
+            else:
+                user_provide_word(userStartingLetter, userWords)
+                break
```

Comment:
Possible simplification...

```suggestion
        success = computer_guess_words(chosenCategory, userStartingLetter, usedWords)
        if not success:
            user_provide_word(userStartingLetter, userWords)
```

Reply:
This was really helpful. Thank you